gst/nnstreamer/tensor_query/tensor_query_sink.h

```diff
@@ -0,0 +1,72 @@
+/**
+ * GStreamer
+ * Copyright (C) 2005 Thomas Vander Stichele <thomas@apestaart.org>
+ * Copyright (C) 2005 Ronald S. Bultje <rbultje@ronald.bitfreak.net>
+ * Copyright (C) 2018 Samsung Electronics Co., Ltd.
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Library General Public
+ * License as published by the Free Software Foundation;
+ * version 2.1 of the License.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Library General Public License for more details.
+ */
+
+/**
+ * Copyright (C) 2021 Junhwan Kim <jejudo.kim@samsung.com>
+ * 
+ * @file    tensor_query_sink.h
+ * @date    09 Jul 2021
+ * @brief   GStreamer plugin to handle tensor query sink
+ * @author  Junhwan Kim <jejudo.kim@samsung.com>
+ * @see     http://github.com/nnstreamer/nnstreamer
+ * @bug     No known bugs
+ * 
+ */
+#ifndef __GST_TENSOR_QUERY_SINK_H__
+#define __GST_TENSOR_QUERY_SINK_H__
+
+#include <gst/gst.h>
+#include <gst/base/gstbasesink.h>
+#include <tensor_common.h>
+
+G_BEGIN_DECLS
+
+
+#define GST_TYPE_TENSOR_QUERY_SINK \
+  (gst_tensor_query_sink_get_type())
+#define GST_TENSOR_QUERY_SINK(obj) \
+  (G_TYPE_CHECK_INSTANCE_CAST((obj),GST_TYPE_TENSOR_QUERY_SINK,GstTensorQuerySink))
+#define GST_TENSOR_QUERY_SINK_CLASS(klass) \
+  (G_TYPE_CHECK_CLASS_CAST((klass),GST_TYPE_TENSOR_QUERY_SINK,GstTensorQuerySinkClass))
+#define GST_IS_TENSOR_QUERY_SINK(obj) \
+  (G_TYPE_CHECK_INSTANCE_TYPE((obj),GST_TYPE_TENSOR_QUERY_SINK))
+#define GST_IS_TENSOR_QUERY_SINK_CLASS(klass) \
+  (G_TYPE_CHECK_CLASS_TYPE((klass),GST_TYPE_TENSOR_QUERY_SINK))
+
+typedef struct _GstTensorQuerySink GstTensorQuerySink;
+typedef struct _GstTensorQuerySinkClass GstTensorQuerySinkClass;
+
+/**
+ * @brief GstTensorQuerySink data structure.
+ */
+struct _GstTensorQuerySink
+{
+  GstBaseSink element; /**< parent object */
```

Comment:
This should be changed to `GstMultiSocketSink` for a 1:N connection later.
It looks good to start with `GstBaseSink`.